website/functions/get-discourse-comments.js

```diff
@@ -0,0 +1,154 @@
+const axios = require('axios')
+
+const { DISCOURSE_DEVBLOG_API_KEY , DISCOURSE_USER_SYSTEM } = process.env
+const DEVBLOG_URL = 'https://docs.getdbt.com/blog/'
+
+// Set API endpoint and headers
+let discourse_endpoint = `https://discourse.getdbt.com`
+let headers = {
+  'Accept': 'application/json',
+  'Api-Key': DISCOURSE_DEVBLOG_API_KEY,
+  'Api-Username': DISCOURSE_USER_SYSTEM,
+}    
+
+async function getDiscourseComments( event ) {
+  try {
+    let postTitle = event.queryStringParameters.title.trim()
+    let postSlug = event.queryStringParameters.slug
+    let postTitleEncoded = encodeURIComponent(postTitle)
+
+    if(!postTitle) throw new Error('Unable to query Discourse API.')
+
+    let topics = await searchDiscourseTopics(postTitleEncoded)
+    
+    let allTopics = topics
+    let allTopicTitles = []
+    let topicExists = false
+    
+    let topicId = null
+    let comments = []
+
+    // Return all the topic titles from Discourse
+    if(topics && topics.length > 0 ) {
+        allTopicTitles = allTopics.reduce((topicsArr, topic) => {
+            topicsArr.push(topic.title)
+            return topicsArr
+        }, [])
+
+        // Check to see if the Dev Blog post title matches any of the Discourse topic titles
+        topicExists = allTopics.some(topic => topic.title === postTitle)
+    }
+
+    console.log('topicExists', topicExists)
+
+    // If it does not exist in Discourse, create a new topic
+    if(!topicExists) {
+        await createDiscourseTopic(postTitle, postSlug)
+
+        topics = await searchDiscourseTopics(postTitleEncoded)
+
+        if(topics && topics?.length > 0 ) {
+            topicId = await getTopicId(topics, postTitle)
+        }
+
+        if(topicId) {
+            comments = await getDiscourseTopicbyID(topicId)
+        }
+     
+        // Remove the the first post of the comments array since it is not a comment
+        if (comments.length) {
+            comments.shift()
+            comments = { topicId, comments }
+        }
+
+        return await returnResponse(200, comments)
+    } else {
```

Comment:
Just a❔for when `!topicExists`. Does this flow create a new topic, then search the api to get that newly created `topicId`, which is used to get the comments from there?

If so, i'm wondering if the discourse create topic endpoint happens to return the new topic id when created? Would it be possible to pull the `topicId` directly from the `await createDiscourseTopic(postTitle, postSlug)` response, and pass it into `getDiscourseTopicbyID(topicId)` from there? That might allow lines 48-52 to be eliminated, but not sure if possible, or if those lines are required for anything else.

Reply:
Yes and great call returning the topicId on the `createDiscourseTopic` instead of having another function ran.

Just pushed a change to return topicId from `createDiscourseTopic` and also updated the Discourse category ID to be a constant.